spec/features/orders/user_cancel_spec.rb

```diff
@@ -0,0 +1,150 @@
+                              )
+
+    @merchant_user2 = @dog_shop.users.create!(name: "Jeremiah",
+                                street_address: "7777 Rastafari Way",
+                                city: "Jamaica",
+                                state: 'FL',
+                                zip_code: 46766,
+                                email: "ziggymarley@example.com",
+                                password: "irie333",
+                                role: 1
+                              )
+
+    @walter = User.create!(name: "Walter White",
+                                 street_address: "6230 Bluerock Ln",
+                                 city: "Albuquerque",
+                                 state: 'NM',
+                                 zip_code: 44565,
+                                 email: "heisenberg@example.com",
+                                 password: "method3",
+                                 role: 0
+                                  )
+  end
+  it "all merchants can fulfill items on an order" do
+    visit '/login'
+
+    fill_in 'email', with: @walter.email
+    fill_in 'password', with: @walter.password
+
+    click_on 'Log In'
+
+    click_on("All Items")
+
+    click_on("GatorSkins")
+    click_on("Add To Cart")
+
+    click_on("Doggy Bone")
+    click_on("Add To Cart")
+
+    click_on("Pully Toy")
+    click_on("Add To Cart")
+
+    click_on("Pully Toy")
+    click_on("Add To Cart")
+
+    visit "/cart"
+
+    click_on("Checkout")
+
+    fill_in :name, with: @walter.name
+    fill_in :address, with: @walter.street_address
+    fill_in :city, with: @walter.city
+    fill_in :state, with: @walter.state
+    fill_in :zip, with: @walter.zip_code
+
+    click_button("Create Order")
+
+    order = Order.last
+
+    visit '/profile/orders'
+
+    click_on(order.id)
+    click_on("Cancel Order")
+
+    expect(current_path).to eq("/profile")
+
+    expect(page).to have_content("Your order has been cancelled.")
+
+    click_on("My Orders")
+
+    expect(page).to have_content("Current status: cancelled")
+  end
+  it "when an order has items that have been fulfilled and that order is cancelled, those items inventories are returned" do
+
+
+    visit '/login'
+
+    fill_in 'email', with: @walter.email
+    fill_in 'password', with: @walter.password
+
+    click_on 'Log In'
+
+    click_on("All Items")
+
+    click_on("GatorSkins")
+    click_on("Add To Cart")
+
+    click_on("Doggy Bone")
+    click_on("Add To Cart")
+
+    click_on("Pully Toy")
+    click_on("Add To Cart")
+
+    click_on("Pully Toy")
+    click_on("Add To Cart")
+
+    visit "/cart"
+
+    click_on("Checkout")
+
+    fill_in :name, with: @walter.name
+    fill_in :address, with: @walter.street_address
+    fill_in :city, with: @walter.city
+    fill_in :state, with: @walter.state
+    fill_in :zip, with: @walter.zip_code
+
+    click_button("Create Order")
+
+    order = Order.last
+
+    order.item_orders.last.update(status: "fulfilled")
```

Comment:
here I would also test the other `items` in the `order` before and after the `order` is cancelled